### ISSUE TITLE
PMIX error out of resource

### ISSUE THREAD

**Issue by jviquerat** [openpmix/openpmix#1513]
Hi,

I am working with a C++ CFD code that uses openmpi. In several cases, I have to call this code multiple times (hundreds to thousands of times) via a single python process, without exiting python.
In this case, I am consistently facing the following error after a fixed amount of calls to the CFD code (994 times) : 

```
PMIX ERROR: OUT-OF-RESOURCE in file src/server/pmix_server_listener.c at line 273
```

I believe this is related to some buffer that gets filled at each call and is eventually full, but I have trouble finding informations about this. I would be very grateful for any information you would have about this problem.

**Comment by dsikich** [comment]
@jviquerat I'm fairly new to this project and the creator of the project is on vacation through December so I will try to assist if I can, but I'm still learning the codebase. I looked and am unable to find this file in either the ompi PMIx code or the PMIx codebase. I believe the host server (ompi) would also be where the listener function would actually be implemented. Do you have this file? I'm not sure where to find it. It might also be worth asking/alerting the ompi team about this issue you are seeing since they may be able to help too.

**Comment by jviquerat** [comment]
@dsikich thanks for your answer. Finally found the file in a previous revision (it is not present anymore in more recent versions) : https://github.com/openpmix/openpmix/blob/858c378719669ee772b5326c3f2d2606ad0c9bd6/src/server/pmix_server_listener.c

**Comment by dsikich** [comment]
@jviquerat yeah this code is in another file in newer versions. The OUT-OF-RESOURCE error is coming from EMFILE, which means there are too many open files, not sure if that helps. What kind of environment and/or machine are you running this on, and how many processes is the C++ using? Is this all on one node?

**Comment by mike0042** [comment]
What is your the open file limit (ulimit -n) in your session set to? 1024 is a common default, so you may need to increase it.

**Comment by rhc54** [comment]
What version of PMIx are you using? I believe that code was revamped in the 2.x series, so it must be something fairly old. Any chance you could update?

**Comment by jviquerat** [comment]
Hi,

Sorry for the delayed answer.
I tried raising the ulimit when the error first came up, to no avail. Then I realized that, for some unknown reason (possibly a local configuration), the python process was not inheriting the environment ulimit value.
Following this solved the problem : https://coderwall.com/p/ptq7rw/increase-open-files-limit-and-drop-privileges-in-python

So someone in my process opens files without closing them, but using this workaround is fine for me. Thank you anyway for your time !
